### PR TITLE
[RT-703] Add documentation for container agent custom token secret

### PR DESCRIPTION
**Pull request by Stuart133**
# Description

Add documentation for the custom token secret feature in the container agent helm chart

# Reasons

Documentation supporting this feature: https://circleci.atlassian.net/browse/RT-703

# Content Checklist
Please follow our style when contributing to CircleCI docs. Our style guide is here: [https://circleci.com/docs/style/style-guide-overview](https://circleci.com/docs/style/style-guide-overview).

Please take a moment to check through the following items when submitting your PR (this is just a guide so will not be relevant for all PRs) 😸:

- [ ] Break up walls of text by adding paragraph breaks.
- [ ] Consider if the content could benefit from more structure, such as lists or tables, to make it easier to consume.
- [ ] Keep the title between 20 and 70 characters.
- [ ] Consider whether the content would benefit from more subsections (h2-h6 headings) to make it easier to consume.
- [ ] Check all headings h1-h6 are in sentence case (only first letter is capitalized).
- [ ] Is there a "Next steps" section at the end of the page giving the reader a clear path to what to read next?
- [ ] Include relevant backlinks to other CircleCI docs/pages.

### CODE REVIEW

File: jekyll/_cci2/container-runner.adoc [hunk 1]
```diff
@@ -145,6 +145,33 @@ agent:
           - name: "other"
 ```
 
+[#custom-secret]
+=== Custom Token Secret
```
Comment (rosieyohannan) [inline jekyll/_cci2/container-runner.adoc:149]:
```suggestion
=== Custom token secret
```

File: jekyll/_cci2/container-runner.adoc [hunk 2]
```diff
@@ -145,6 +145,33 @@ agent:
...
+
+Using the above configuration will provision a Kubernetes secret containing your Resource Class tokens. In some circumstances you may wish to provision your own secret, or you simply might not want to specify the tokens via helm. Instead you can provision your own Kubernetes secret containing your tokens and specify its name in the `agent.customSecret` field.
```
Comment (rosieyohannan) [inline jekyll/_cci2/container-runner.adoc:151]:
```suggestion
Using the configuration described above provisions a Kubernetes secret containing your resource class tokens. In some circumstances, you may wish to provision your own secret, or you simply might not want to specify the tokens via helm. Instead, you can provision your own Kubernetes secret containing your tokens and specify its name in the `agent.customSecret` field.
```

File: jekyll/_cci2/container-runner.adoc [hunk 3]
```diff
@@ -145,6 +145,33 @@ agent:
...
+
+The secret should contain a field for each Resource Class, using the Resource Class name as the key and the token as the value. Consider the following `resourceClasses` configuration:
```
Comment (rosieyohannan) [inline jekyll/_cci2/container-runner.adoc:153]:
```suggestion
The secret should contain a field for each resource class, using the Resource Class name as the key and the token as the value. Consider the following `resourceClasses` configuration:
```

File: jekyll/_cci2/container-runner.adoc [hunk 4]
```diff
@@ -145,6 +145,33 @@ agent:
...
+
+```yaml
+agent:
+  resourceClasses:  
+    circleci-runner/resourceClass:
+      metadata:
+        annotations:
+          custom.io: my-annotation
+    
+    circleci-runner/resourceClass2:
```
Comment (rosieyohannan) [inline jekyll/_cci2/container-runner.adoc:163]: For values that the reader needs to substitute for their own data, we use `<my-token>`, <my-resource-class> or whatever is appropriate. Could you update this snippet to use this format?

File: jekyll/_cci2/container-runner.adoc [hunk 5]
```diff
@@ -145,6 +145,33 @@ agent:
...
+```
+The corresponding custom secret would have 2 fields:
```
Comment (rosieyohannan) [inline jekyll/_cci2/container-runner.adoc:165]:
```suggestion

The corresponding custom secret would have 2 fields:
```

File: jekyll/_cci2/container-runner.adoc (hunk 4 above)
Comment (rosieyohannan) [reply jekyll/_cci2/container-runner.adoc:163]: Oh and the one below to match

File: jekyll/_cci2/container-runner.adoc [hunk 6]
```diff
@@ -145,6 +145,33 @@ agent:
...
+
+```yaml
+circleci-runner.resourceClass: TOKEN1
+circleci-runner.resourceClass2: TOKEN2
+```
+
+Due to Kubernetes secret key character constraints, the `/` separating the Namespace and Resouce Class name is replaced with a `.` character. Other than this the name must exactly match the `resourceClasses` config to match the token with the correct configuration.
```
Comment (rosieyohannan) [inline jekyll/_cci2/container-runner.adoc:172]:
```suggestion
Due to Kubernetes secret key character constraints, the `/` separating the namespace and resouce class name is replaced with a `.` character. Other than this, the name must exactly match the `resourceClasses` config to match the token with the correct configuration.
```

File: jekyll/_cci2/container-runner.adoc [hunk 7]
```diff
@@ -145,6 +145,33 @@ agent:
...
+
+Even if there is no further pod configuration, the resource class must be present in `resourceClasses` as an emtpy map, as shown by `circleci-runner/resourceClass2` in the above config
```
Comment (rosieyohannan) [inline jekyll/_cci2/container-runner.adoc:174]:
```suggestion
Even if there is no further pod configuration, the resource class must be present in `resourceClasses` as an emtpy map, as shown by `circleci-runner/resourceClass2` in the above config example.
```

File: jekyll/_cci2/container-runner.adoc [hunk 8]
```diff
@@ -167,8 +194,12 @@ The following are **CircleCI specific settings**:
 | `container-agent` (the name of the deployment)
 
 | agent.resourceClasses *Default must be updated in order to run a job successfully*
-| Resource class task configuration. See "Resource Class Configuration” section below
-| " "
+| Resource class task configuration. See "<<resource-class-configuration-custom-pod,Resource Class Configuration>>" section above
+| {}
+
+| agent.customSecret
+| A user provided Kubernetes containing resource class tokens. See "<<custom-secret,Custom Token Secret>>" section above
```
Comment (rosieyohannan) [inline jekyll/_cci2/container-runner.adoc:201]:
```suggestion
| Resource class task configuration. See the "<<resource-class-configuration-custom-pod,Resource Class Configuration>>" section above.
| {}

| agent.customSecret
| A user provided Kubernetes containing resource class tokens. See the "<<custom-secret,Custom Token Secret>>" section above.
```

**Review by rosieyohannan — COMMENTED** [review]
Thank you!

**Comment by sebastian-lerner** [comment]
LGTM thanks stuart

**Comment by sebastian-lerner** [comment]
@rosieyohannan can you help us approve and merge when you get a chance please?

**Review by rosieyohannan — APPROVED** [review]

**Pull request merged**